official/utils/logging/hooks.py

```diff
@@ -0,0 +1,115 @@
+               every_n_steps=None,
+               every_n_secs=None,
+               warm_steps=0):
+    """Initializer for ExamplesPerSecondHook.
+
+    Args:
+      batch_size: Total batch size used to calculate examples/second from
+        global time.
+      every_n_steps: Log stats every n steps.
+      every_n_secs: Log stats every n seconds. Exactly one of the
+      `every_n_steps` or `every_n_secs` should be set.
+      warm_steps: skip this number of steps before logging and running
+        average. warm_steps steps refers to global steps across all workers,
+        not on each worker
+
+    Raises:
+      ValueError: if neither `every_n_steps` or `every_n_secs` is set, or
+      both are set.
+    """
+
+    if (every_n_steps is None) == (every_n_secs is None):
+      raise ValueError('exactly one of every_n_steps'
+                       ' and every_n_secs should be provided.')
+
+    self._timer = tf.train.SecondOrStepTimer(
+        every_steps=every_n_steps, every_secs=every_n_secs)
+
+    self._step_train_time = 0
+    self._total_steps = 0
+    self._batch_size = batch_size
+    self._warm_steps = warm_steps
+
+  def begin(self):
+    """Called once before using the session to check global step."""
+    self._global_step_tensor = tf.train.get_global_step()
+    if self._global_step_tensor is None:
+      raise RuntimeError(
+          'Global step should be created to use StepCounterHook.')
+
+  def before_run(self, run_context):  # pylint: disable=unused-argument
+    """Called before each call to run().
+
+    Args:
+      run_context: A SessionRunContext object.
+
+    Returns:
+      A SessionRunArgs object or None if never triggered.
+    """
+    return tf.train.SessionRunArgs(self._global_step_tensor)
+
+  def after_run(self, run_context, run_values):  # pylint: disable=unused-argument
+    """Called after each call to run().
+
+    Args:
+      run_context: A SessionRunContext object.
+      run_values: A SessionRunValues object.
+    """
+    global_step = run_values.results
+
+    if self._timer.should_trigger_for_step(
+        global_step) and global_step > self._warm_steps:
+      elapsed_time, elapsed_steps = self._timer.update_last_triggered_step(
+          global_step)
+      if elapsed_time is not None:
+        self._step_train_time += elapsed_time
+        self._total_steps += elapsed_steps
+
+        # average examples per second is based on the total (accumulative)
+        # training steps and training time so far
+        average_examples_per_sec = self._batch_size * (
+            self._total_steps / self._step_train_time)
+        # current examples per second is based on the elapsed training steps
+        # and training time per batch
+        current_examples_per_sec = self._batch_size * (
+            elapsed_steps / elapsed_time)
+        # Current examples/sec followed by average examples/sec
+        tf.logging.info('Batch [%g]:  current exp/sec = %g, average exp/sec = '
```

Comment:
For now, we can probably stick with the standard set in https://github.com/tensorflow/tensorflow/blob/r1.6/tensorflow/python/training/basic_session_run_hooks.py#L244 -- if we end up with a use case for more granularity, we can always go back and add, but we tend to always be in danger of too many options rather than too few.